plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java

```diff
@@ -47,41 +53,114 @@ public HudsonPrivateSecurityRealmConfigurator() {
     }
 
     private static Collection<UserWithPassword> getter(HudsonPrivateSecurityRealm target) {
         return target.getAllUsers().stream()
-                .map(u -> new UserWithPassword(u.getId(), null)) // password isn't actually stored, only hashed
+                .map(u -> {
+                    UserWithPassword user = new UserWithPassword(u.getId(), null);
+                    user.setName(u.getFullName());
+                    user.setDescription(u.getDescription());
+                    List<UserProperty> properties = u.getAllProperties()
+                        .stream()
+                        .filter(userProperty -> !userProperty.getClass().getName().equals("com.cloudbees.plugins.credentials.UserCredentialsProvider$UserCredentialsProperty"))
+                        .collect(Collectors.toList());
+                    user.setProperties(properties);
+
+                    return user;
+                })
                 .collect(Collectors.toList());
     }
 
     private static void setter(HudsonPrivateSecurityRealm target, Collection<UserWithPassword> value) throws IOException {
         for (UserWithPassword user : value) {
+            User updatedUser = createAccountOrLookupById(target, user);
+            updatedUser.setFullName(user.name);
+            updatedUser.setDescription(user.description);
+            if (user.getProperties() != null) {
+                for (UserProperty property : user.getProperties()) {
+                    updatedUser.addProperty(property);
+                }
+            }
+        }
+    }
+
+    private static User createAccountOrLookupById(HudsonPrivateSecurityRealm target, UserWithPassword user)
+        throws IOException {
+        User updatedUser;
+        if (StringUtils.isNotBlank(user.password)) {
             if (StringUtils.startsWith(user.password, HASHED_PASSWORD_PREFIX)) {
                 try {
-                    target.createAccountWithHashedPassword(user.id, user.password);
-                } catch (IllegalArgumentException e) {
-                    logger.log(Level.WARNING, "Failed to create user with presumed hashed password", e);
+                    updatedUser = target
+                        .createAccountWithHashedPassword(user.id, user.password);
+                } catch (IllegalArgumentException | IOException e) {
+                    logger.log(Level.WARNING,
+                        "Failed to create user with presumed hashed password", e);
                     // fallback, just create the account as is
-                    target.createAccount(user.id, user.password);
+                    updatedUser = target.createAccount(user.id, user.password);
                 }
             } else {
-                target.createAccount(user.id, user.password);
+                updatedUser = target.createAccount(user.id, user.password);
+            }
+        } else {
+            updatedUser = User.getById(user.id, false);
+            if (updatedUser == null) {
```

Comment:
I'm wondering if it makes sense to create the account if it doesn't exist without a password